hazelcast/src/main/java/com/hazelcast/core/ICacheManager.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.core;
+
+import com.hazelcast.cache.ICache;
+
+/**
+ * {@code ICacheManager} is the entry point to access JSR-107 (JCache) caches via {@link HazelcastInstance} interface.
+ * Hazelcast's {@code ICacheManager} provides access to JCache caches configured cluster-wide, even when created by different
+ * JCache {@link javax.cache.CacheManager}s.
+ *
+ * <p>Note that this interface is not related to JCache {@link javax.cache.CacheManager}.</p>
```

Comment:
Imho it would be good to give some background information why this class exists.